fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1031,6 +1033,11 @@ ThreadResult<RangeResult> MultiVersionTransaction::readBlobGranules(const KeyRan
                                                                     Version beginVersion,
                                                                     Optional<Version> readVersion,
                                                                     ReadBlobGranuleContext granuleContext) {
+	if (MultiVersionApi::isNetworkThread()) {
+		// Check to avoid synchronous ThreadResult call when already on another main thread, which would cause a
```

Comment:
Normally the API functions are called from client threads. Why do you expect it to be called from the main thread and what kind of deadlock may occur here? Does it make a difference if the CALLBACKS_ON_EXTERNAL_THREADS option is set?

If you need this kind of check, let's better use the existing g_network->isOnMainThread method

Reply:
@ajbeamon correct me if i am wrong, but the distinction here was if someone wants to do a readBlobGranules as a callback on a future on one network thread, but run the callback on an external thread.
In that case, since there is only one main thread, the isOnMainThread wouldn't address the problem, since this fix is trying to prevent a multi-thread deadlock